specification/resources/resource-manager/Microsoft.Resources/stable/2021-04-01/resources.json

```diff
@@ -4605,6 +4605,50 @@
       ],
       "description": "Deployment properties."
     },
+    "NestedDeploymentParameter": {
+      "properties": {
+        "value": {
+          "type": "object",
+          "description": "Input value to the parameter."
+        },
+        "reference": {
+          "$ref": "#/definitions/KeyVaultParameterReference",
+          "description": "Azure Key Vault parameter reference."
+        },
+      },
+      "required": [
```

Comment:
How to express `either` aka `oneOf`?

Reply:
Technically this is valid: https://github.com/Azure/azure-resource-manager-schemas/blob/ec38aad589d0a30ea09b4dca062ed495c5daa0cd/schemas/2019-04-01/deploymentParameters.json#LL48C1-L55C8

That being said, I would probably recommend against using it here, because these files are used to generate SDKs & other tools - and I'm not sure whether this is going to work well with existing tooling.

Reply:
Ok, won't touch then.